### PR TITLE
Remove support for comma separated strings

### PR DESCRIPTION
**Pull request by BraisGabin**
We did this change too long ago. Now on 2.0 we can remove the support for this and only support lists.

### CODE REVIEW

**Review by schalkms — APPROVED** [review]
Feels good to throw away some code and thus reduce complexity overall!

File: detekt-rules-exceptions/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocationSpec.kt [hunk 1]
```diff
@@ -102,17 +102,4 @@ class ExceptionRaisedInUnexpectedLocationSpec {
         )
         assertThat(findings).hasSize(1)
     }
-
-    @Test
-    fun `reports the configured method with String`() {
-        val config = TestConfig("methodNames" to "toDo,todo2")
```
Comment (schalkms) [inline detekt-rules-exceptions/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocationSpec.kt:108]: I assume we already have a test for `list(toDo,todo2)` or something similar, right?

File: detekt-rules-libraries/src/test/kotlin/io/gitlab/arturbosch/detekt/libraries/ForbiddenPublicDataClassSpec.kt [hunk 2]
```diff
@@ -153,16 +154,4 @@ class ForbiddenPublicDataClassSpec {
         val config = TestConfig("ignorePackages" to listOf("*.hello", "com.example"))
         assertThat(ForbiddenPublicDataClass(config).compileAndLint(code)).isEmpty()
     }
-
-    @Test
-    fun `public data class inside an ignored package should pass config as string`() {
```
Comment (schalkms) [inline detekt-rules-libraries/src/test/kotlin/io/gitlab/arturbosch/detekt/libraries/ForbiddenPublicDataClassSpec.kt:158]: Same here, right?

File: detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ForbiddenClassNameSpec.kt [hunk 3]
```diff
@@ -31,19 +31,6 @@ class ForbiddenClassNameSpec {
         ).hasSize(1)
     }
 
-    @Test
-    fun `should report classes with forbidden names using config string`() {
```
Comment (schalkms) [inline detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ForbiddenClassNameSpec.kt:35]: Same here, right?

**Review by marschwar — APPROVED** [review]
You keep coming up with nice improvements. :1st_place_medal:

File: detekt-rules-exceptions/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocationSpec.kt (hunk 1 above)
Comment (BraisGabin) [reply detekt-rules-exceptions/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocationSpec.kt:108]: We do, it's the one above this one.

File: detekt-rules-libraries/src/test/kotlin/io/gitlab/arturbosch/detekt/libraries/ForbiddenPublicDataClassSpec.kt (hunk 2 above)
Comment (BraisGabin) [reply detekt-rules-libraries/src/test/kotlin/io/gitlab/arturbosch/detekt/libraries/ForbiddenPublicDataClassSpec.kt:158]: Same, above this one.

File: detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ForbiddenClassNameSpec.kt (hunk 3 above)
Comment (BraisGabin) [reply detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ForbiddenClassNameSpec.kt:35]: Same, but this time 2 above. I checked all of these manually and remove the repetead one, and edit the other ones.

File: detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/config/validation/InvalidPropertiesConfigValidatorSpec.kt [hunk 4]
```diff
@@ -103,8 +103,8 @@ internal class InvalidPropertiesConfigValidatorSpec {
         assertThat(result).anySatisfy { notification ->
             assertThat(notification.message)
                 .contains("style>MagicNumber>ignoreNumbers")
-                .contains("should be a YAML array instead of a comma-separated String")
-            assertThat(notification.level).isEqualTo(Notification.Level.Warning)
+                .contains("should be a YAML array instead of a String")
+            assertThat(notification.level).isEqualTo(Notification.Level.Error)
```
[the remark this answers is not in the document]
Comment (BraisGabin) [reply detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/config/validation/InvalidPropertiesConfigValidatorSpec.kt:107]: Done

File: detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/config/validation/ValidateConfigSpec.kt [hunk 5]
```diff
@@ -54,10 +54,9 @@ internal class ValidateConfigSpec {
 
         val result = validateConfig(config, baseline, emptySet())
 
-        val expectedLevel = if (warningsAsErrors) Notification.Level.Error else Notification.Level.Warning
         assertThat(result).anySatisfy { notification ->
             assertThat(notification.message).contains("style>MagicNumber")
-            assertThat(notification.level).isEqualTo(expectedLevel)
+            assertThat(notification.level).isEqualTo(Notification.Level.Error)
```
[the remark this answers is not in the document]
Comment (BraisGabin) [reply detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/config/validation/ValidateConfigSpec.kt]: Good catch! I adapted the test to keep testing what it was meant too as you described.

**Pull request merged**